File: official/docs/node/end_shipper/update_endshipper.js
const Easypost = require('@easypost/api');
const api = new Easypost('<YOUR_TEST/PRODUCTION_API_KEY>');

const endShipper = api.EndShipper.retrieve('es_...');

endShipper.name = 'NEW NAME';
endShipper.company = 'BAZ';
endShipper.street1 = '164 TOWNSEND STREET UNIT 1';
endShipper.street2 = 'UNIT 1';
endShipper.city = 'SAN FRANCISCO';
endShipper.state = 'CA';
endShipper.zip = '94107';
endShipper.country = 'US';
endShipper.phone = '555-555-5555';
endShipper.email = 'FOO@EXAMPLE.COM';

endShipper.save().then(console.log);

File: official/docs/java/endShipper/UpdateEndShipper.java
import com.easypost.EasyPost;
import java.util.HashMap;

public class UpdateEndShipper {
    public static void main(String[] args) {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");

        EndShipper endShipper = EndShipper.retrieve("es_...");

        HashMap<String, Object> updateParams = new HashMap<String, Object>();

        updateParams.put("name", "NEW NAME");
        updateParams.put("company", "BAZ");
        updateParams.put("street1", "164 TOWNSEND STREET UNIT 1");
        updateParams.put("street2", "UNIT 1");
        updateParams.put("city", "SAN FRANCISCO");
        updateParams.put("state", "CA");
        updateParams.put("zip", "94107");
        updateParams.put("country", "US");
        updateParams.put("phone", "555-555-5555");
        updateParams.put("email", "FOO@EXAMPLE.COM");

        EndShipper updatedEndShipper = endShipper.update(updateParams);

        System.out.println(updatedEndShipper);
    }
}
